pkg/network/setup/network.go

```diff
@@ -53,12 +53,17 @@ func (v VMNetworkConfigurator) getPhase1NICs(launcherPID *int, networks []v1.Net
 	var nics []podNIC
 
 	for i := range networks {
-		// SR-IOV devices are not part of the phases.
-		if iface := vmispec.LookupInterfaceByName(v.vmi.Spec.Domain.Devices.Interfaces, networks[i].Name); iface.SRIOV != nil {
+		iface := vmispec.LookupInterfaceByName(v.vmi.Spec.Domain.Devices.Interfaces, networks[i].Name)
+		if iface == nil {
```

Comment:
I think this check is redundant, there are validations for network-interfaces alignment if there is an interface and no corresponding network we will probably have bigger problems to worry about :sweat_smile:

Reply:
Yes, there are protections before the code gets to this point, but still, I don't feel comfortable to remove this check.
If there will be a bug, the check if the interface has an SR-IOV binding will panic.

I don't think it hurts to keep this little`if` statement.

Reply:
It was moved up the stack from `newPodNIC`.

Reply:
This seems to me like a standoff. On one side, this was not checked before here and all was fine.
The change embraced the pedantic approach of checking it as it existed in `newPodNIC`.

Personally, I would have embraced the more loose approach and depend on the configuration being correct. If there is risk, I would just make sure to check this at some entry point to avoid re-checking later.

But as both approaches existed, it is less trivial to decide what to do.

Reply:
@AlonaKaplan what do you think should be done here?

Reply:
As the check `if correspondingNetworkIface == nil {` was removed from `newPodNIC` it makes sense to have it here.